### PR TITLE
pkp/pkp-lib#7523 Can save DOI settings with no suffix selected

### PR DESCRIPTION
**Pull request by ewhanson**

### CODE REVIEW

File: classes/submission/Repository.php [hunk 1]
```diff
@@ -26,9 +26,10 @@
 use Illuminate\Support\Enumerable;
 use Illuminate\Support\LazyCollection;
 use PKP\context\Context;
+use PKP\context\PKPSection;
```
Comment (bozana) [inline classes/submission/Repository.php:29]: This class should not exist any more, after the conversion to the new Repository/DAO/Map pattern.
I see also some old paths in OJS... They will probably go away once you rebase again, but be careful if there are any conflicts... :-)

File: classes/doi/exceptions/DoiException.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DoiException
+ *
+ * @brief Base class to catch for all DOI-related exceptions
+ */
+
+namespace PKP\doi\exceptions;
+
+class DoiException extends \Exception
```
Comment (bozana) [inline classes/doi/exceptions/DoiException.php:17]: Hmmm... Why do we need this empty class?

**Review by bozana — COMMENTED** [review]
just a question :-)

File: classes/submission/Repository.php (hunk 1 above)
Comment (ewhanson) [reply classes/submission/Repository.php:29]: Not sure why that was added, but I've rebased and this use statement was the only reference to it. I've removed it.

File: classes/doi/exceptions/DoiException.php (hunk 2 above)
Comment (ewhanson) [reply classes/doi/exceptions/DoiException.php:17]: The idea was to create a base class for DOI-related exceptions, so they could be caught, but looking again, I can refactor the existing `DoiActionException` to work in all cases so we won't need an empty base class.

**Pull request merged**